lib/app.js

```diff
@@ -41,8 +43,29 @@ if (otpConfig.customIcons) {
   })
 }
 
-if (jsConfig.applyCustomIconLogic) {
-  jsConfig.applyCustomIconLogic(customIcons)
+const BaseModeIcon = jsConfig.BaseModeIcon || ClassicModeIcon
```

Comment:
I'm not sure we need to import ClassicModeIcon and default to it if BaseModeIcon is not provided. Let's just isolate all of the icon config to `config.js`.